src/StackExchange.Redis/ConfigurationOptions.cs

```diff
@@ -296,15 +296,26 @@ private static RemoteCertificateValidationCallback TrustIssuerCallback(X509Certi
         {
             if (issuer == null) throw new ArgumentNullException(nameof(issuer));
 
-            return (object _, X509Certificate? certificate, X509Chain? __, SslPolicyErrors sslPolicyError)
-                => sslPolicyError == SslPolicyErrors.RemoteCertificateChainErrors
-                    && certificate is X509Certificate2 v2
-                    && CheckTrustedIssuer(v2, issuer);
+            return (object _, X509Certificate? certificate, X509Chain? __, SslPolicyErrors sslPolicyError) =>
+            {
+                // If we're already valid, there's nothing further to check
+                if (sslPolicyError == SslPolicyErrors.None)
+                {
+                    return true;
```

Comment:
Hm. Based on the documentation I did not _think_ the intention was pinning

https://github.com/StackExchange/StackExchange.Redis/blob/441f89afeff6d9a55ecf6e413ed5bd29b5aec19d/src/StackExchange.Redis/ConfigurationOptions.cs#L282

But a good question to ask.

Reply:
Since @vcsjones and I have had a discussion offline regarding what that sentence means... if it is pinning, the code's wrong.  If it isn't pinning, the docs would probably be more clear by replacing "even if" with "when".

* I think "even if" means "when normal, and when also", which would make it pinning.
* @vcsjones agrees, but thinks "even if not" is somehow special. Which I disagree with.

/me and @vcsjones now watch to see what Nick says the right answer should be.

Reply:
Playing favorites only gets me in trouble, tried going with making everyone angry instead - thoughts?

Reply:
So is the intention that anything that's trusted by default is acceptable, and if it's not trusted then (and only then) this anchor is checked as a fallback?

If that's the intention, my concern in the docs is with the word "even";

> Create a certificate validation check that checks against the supplied issuer even when not known by the machine.

Sounds like pinning to me.

> Create a certificate validation check that checks against the supplied issuer when not known by the machine.

Sounds like a fallback.